compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIESplitLogicalObjFifos.cpp

```diff
@@ -348,23 +348,87 @@ void AMDAIESplitLogicalObjFifosPass::runOnOperation() {
     return signalPassFailure();
   }
 
+  // Fetch and store all unique pairs of L2<->L1 Copy ops. This would helps us
+  // figure out the split factor for all LogicalObjectFifos.
+  DenseMap<Operation *, DenseSet<Operation *>> uniqueL2L1Pair;
+  moduleOp->walk([&](Operation *op) -> WalkResult {
```

Comment:
I think this could be put in a utility function. Also, instead of walking all the copy ops, I think you can walk the objectFifos being split and their producers/consumers, there should be existing functions: `getCopyLikeProducers`, `getCopyLikeConsumers`.

Reply:
Good point.

I did try that but it seems to be doing a similar thing which the current `walk` achieves.

For LHS/RHS I need L2 -> L1 (so I need `getCopyLikeConsumers`)
For OUT I need L1 -> L2 (so I need `getCopyLikeProducers`)

Since, currently, there is no way to identify which is what by walking just the objectFifos, I'll have to invoke both the APIs and then filter out L2<->L1.

> I think you can avoid relying on the memory spaces by collecting all copy ops that operate on the objectFifos being split.

In the `walk` which I've implemented, a similar feat is being achieved where I have all the copy ops and I'm filtering L2<->L1. I don't think I can get rid of the memory checks. In case I missed something, let me know.

Reply:
You don't need to filter out the `L2<->L1` connections as you just need to take the other side of the DMA being split: if the producer of an objectFifo and the objectFifo is being split, then you need to retrieve copy-like consumers and the other way around. It doesn't matter whether the producer/consumer is L1<->L2, L3<->L2, L1<->L3 or something else, so we should avoid making that assumption. In general, we should avoid checking memory space hardcoding etc whenever possible as it makes the logic fragile towards future changes (a memory space being added, skipped, removed, renamed).

Reply:
Aah yes! I missed "`instead of walking all the copy ops, I think you can walk the objectFifos being split`".

Makes sense. Updated. Thanks!